src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c

```diff
@@ -354,10 +355,18 @@ static int xocl_preserve_mem(struct xocl_drm *drm_p, struct mem_topology *new_to
 
 static bool xocl_xclbin_in_use(struct xocl_dev *xdev)
```

Comment:
xocl_xclbin_in_use() is a simple function that performs the check, the side effect of inserting an error into some global list is not expected. please move them to outside of this function.

Reply:
updated